### PR TITLE
Feature/GitHub scrape service

### LINKED ISSUE #146: Scrape OpenFisca-aotearoa tests from Github (scenarios)
- [x] Scrape urls for test files
- [x] Scrape tests from raw data url
- [x] add scenarios directory to .gitignore
- [x] write directory structure (scenarios) same as on openfisca-aotearoa

part of #62

### PR DESCRIPTION
**Pull request by dlouise64**
Scraping openfisca-aotearoa .yaml test files.

closes #146

### CODE REVIEW

**Comment by Br3nda** [comment]
needs tests.

File: Gemfile [hunk 1]
```diff
@@ -55,6 +55,8 @@ gem 'jquery-rails'
 # Markdown
 gem 'redcarpet'
 
+gem 'metainspector'
```
Comment (Br3nda) [inline Gemfile]: how about a comment saying what this does, to help future devs doing a gem clean out one day

File: app/services/github_scrape_service.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class GithubScrapeService
+  SCENARIOS_DIR = './app/scenarios'
```
Comment (Br3nda) [inline app/services/github_scrape_service.rb]: how about putting this in `./tmp`

File: app/services/github_scrape_service.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  RAW_FILE_DIR = 'https://raw.githubusercontent.com/ServiceInnovationLab/openfisca-aotearoa/master/'
```
Comment (Br3nda) [inline app/services/github_scrape_service.rb]: > can this be in config? or maybe the branch in config too.

File: app/services/github_scrape_service.rb (hunk 2 above)
Comment (dlouise64) [reply app/services/github_scrape_service.rb]: Good idea :) It's in tmp/scenarios now

File: app/services/github_scrape_service.rb (hunk 3 above)
Comment (dlouise64) [reply app/services/github_scrape_service.rb]: @Br3nda when you mean config, do you mean like `config/development.rb?` or in `.env`

File: app/services/github_scrape_service.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+class GithubScrapeService
```
Comment (dlouise64) [inline app/services/github_scrape_service.rb]: @Br3nda these red lines saying, tests don't exist for them, I've tried writing some. Can you have a look at it when you have time thanks!

File: Gemfile [hunk 5]
```diff
@@ -55,6 +55,9 @@ gem 'jquery-rails'
 # Markdown
 gem 'redcarpet'
 
+# Web scraper
```
Comment (Br3nda) [inline Gemfile]:
```suggestion
# Web scraper (used to fetch Open Fisca tests/scenarios from github)
```

File: app.json [hunk 6]
```diff
@@ -4,6 +4,12 @@
     "postdeploy": "rake db:migrate"
   },
   "env": {
+    "GITHUB_RAW_FILE_PATH": {
+      "required": true
+    },
+    "GITHUB_TESTS_PATH": {
+      "required": true
```
Comment (Br3nda) [inline app.json]:
```suggestion
      "required": true,
      "value": "/tree/master/openfisca_aotearoa/tests"
```

File: app.json [hunk 7]
```diff
@@ -4,6 +4,12 @@
     "postdeploy": "rake db:migrate"
   },
   "env": {
+    "GITHUB_RAW_FILE_PATH": {
+      "required": true
```
Comment (Br3nda) [inline app.json]:
```suggestion
      "value": "https://raw.githubusercontent.com/ServiceInnovationLab/openfisca-aotearoa/master/"
```

File: app.json [hunk 8]
```diff
@@ -22,8 +28,23 @@
     "RAILS_SERVE_STATIC_FILES": {
       "required": true
     },
+    "RAW_FILE_STRIP_URL": {
+      "required": true
```
Comment (Br3nda) [inline app.json]:
```suggestion
      "value": "https://github.com/ServiceInnovationLab/openfisca-aotearoa/blob/master/"
```

File: app.json [hunk 9]
```diff
@@ -22,8 +28,23 @@
...
+    },
+    "SECOND_LEVEL_INCLUDE": {
+      "required": true
```
Comment (Br3nda) [inline app.json]:
```suggestion
      "value": "blob/master/openfisca_aotearoa/tests/"
```

File: env-example [hunk 10]
```diff
@@ -1,6 +1,13 @@
 LANG: en
 OPENFISCA_URL: https://api.rules.nz
 GITHUB_URL: https://github.com/ServiceInnovationLab/openfisca-aotearoa
+GITHUB_TESTS_PATH: /tree/master/openfisca_aotearoa/tests
```
Comment (Br3nda) [inline env-example]: this should be `KEY: VALUE`, not `KEY = VALUE`

File: env-example [hunk 11]
```diff
@@ -1,6 +1,13 @@
...
+GITHUB_RAW_FILE_PATH: https://raw.githubusercontent.com/ServiceInnovationLab/openfisca-aotearoa/master/
+SCENARIOS_DIR: ./tmp/scenarios
+TOP_LEVEL_INCLUDE = tree/master/openfisca_aotearoa/tests/
```
Comment (Br3nda) [inline env-example]: this should be `KEY: VALUE`, not `KEY = VALUE`

File: env-example [hunk 12]
```diff
@@ -1,6 +1,13 @@
...
+SECOND_LEVEL_INCLUDE = blob/master/openfisca_aotearoa/tests/
```
Comment (Br3nda) [inline env-example]: this should be `KEY: VALUE`, not `KEY = VALUE`

File: env-example [hunk 13]
```diff
@@ -1,6 +1,13 @@
...
+SECOND_LEVEL_STRIP_URL = https://github.com/ServiceInnovationLab/openfisca-aotearoa/tree/master/openfisca_aotearoa/tests/
```
Comment (Br3nda) [inline env-example]: this should be `KEY: VALUE`, not `KEY = VALUE`

File: env-example [hunk 14]
```diff
@@ -1,6 +1,13 @@
...
+RAW_FILE_STRIP_URL = https://github.com/ServiceInnovationLab/openfisca-aotearoa/blob/master/
```
Comment (Br3nda) [inline env-example]:
```suggestion
RAW_FILE_STRIP_URL: https://github.com/ServiceInnovationLab/openfisca-aotearoa/blob/master/
```

File: spec/services/github_scrape_service.rb [hunk 15]
```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
```
Comment (Br3nda) [inline spec/services/github_scrape_service.rb]: i think this file needs renaming to github_scrape_service_spec.rb

**Comment by Br3nda** [comment]
I don't think your tests are running, i'm working out why now

**Comment by dlouise64** [comment]
@Br3nda Hey Brenda are you able to approve percy as I dont have access to it, thanks!

**Comment by Br3nda** [comment]
Percy approved.

**Pull request merged**